File: Day08/Hard/Solution.py
class Solution(object):
    def canBeTypedWords(self, text, brokenLetters):
        broken = False
        ans = 0

        for i in range(len(text)):
            if text[i] == ' ':
                if not broken:
                    ans += 1
                broken = False

            for j in range(len(brokenLetters)):
                if text[i] == brokenLetters[j]:
                    broken = True

        if not broken:
            ans += 1

        return ans


if __name__ == "__main__":
    text = "hello world"
    brokenLetters = "ad"

    sol = Solution()
    result = sol.canBeTypedWords(text, brokenLetters)

    print("Number of words that can be typed:", result)

File: Day08/Hard/Solution.java
class Solution {
    public int canBeTypedWords(String text, String brokenLetters) {
        boolean broken = false;
        int ans = 0;

        for (int i = 0; i < text.length(); i++) {
            if (text.charAt(i) == ' ') {
                if (!broken) ans++;
                broken = false;
            }
            for (int j = 0; j < brokenLetters.length(); j++) {
                if (text.charAt(i) == brokenLetters.charAt(j)) {
                    broken = true;
                }
            }
        }

        if (!broken) ans++;
        return ans;        
    }
}

public class Main {
    public static void main(String[] args) {
        String text = "hello world";
        String brokenLetters = "ad";

        Solution sol = new Solution();
        int result = sol.canBeTypedWords(text, brokenLetters);

        System.out.println("Number of words that can be typed: " + result);
    }
}
